diff --git a/tensor.go b/tensor.go
index 12a1caaa..794d00d8 100644
--- a/tensor.go
+++ b/tensor.go
@@ -67,13 +67,8 @@ const (
 	Half              DataType = C.TF_HALF
 	Float8e5m2        DataType = C.TF_FLOAT8_E5M2
 	Float8e4m3fn      DataType = C.TF_FLOAT8_E4M3FN
-	Float8e4m3fnuz    DataType = C.TF_FLOAT8_E4M3FNUZ
-	Float8e4m3b11fnuz DataType = C.TF_FLOAT8_E4M3B11FNUZ
-	Float8e5m2fnuz    DataType = C.TF_FLOAT8_E5M2FNUZ
 	Int4              DataType = C.TF_INT4
 	Uint4             DataType = C.TF_UINT4
-	Int2              DataType = C.TF_INT2
-	Uint2             DataType = C.TF_UINT2
 )
 
 // Tensor holds a multi-dimensional array of elements of a single data type.
@@ -563,7 +558,7 @@ func isTensorSerializable(dataType DataType) error {
 	// serialization and deserialization of Tensors.  Till then capitalize
 	// on knowledge of the implementation for numeric types.
 	switch dataType {
-	case Float, Double, Int32, Uint8, Int16, Int8, Complex, Int64, Bool, Quint8, Qint32, Bfloat16, Qint16, Quint16, Uint16, Complex128, Half, Float8e5m2, Float8e4m3fn, Float8e4m3fnuz, Float8e4m3b11fnuz, Float8e5m2fnuz, Int4, Uint4, Int2, Uint2:
+	case Float, Double, Int32, Uint8, Int16, Int8, Complex, Int64, Bool, Quint8, Qint32, Bfloat16, Qint16, Quint16, Uint16, Complex128, Half, Float8e5m2, Float8e4m3fn, Int4, Uint4:
 		return nil
 	default:
 		return fmt.Errorf("serialization of tensors with the DataType %d is not yet supported, see https://github.com/tensorflow/tensorflow/issues/6003", dataType)
